src/main/kotlin/org/team5419/fault/hardware/ctre/BerkeliumFX.kt

```diff
@@ -0,0 +1,42 @@
+package org.team5419.fault.hardware.ctre
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX
+import org.team5419.fault.math.units.SIUnit
+import org.team5419.fault.math.units.SIKey
+import org.team5419.fault.math.units.Meter
+import org.team5419.fault.math.units.Second
+import org.team5419.fault.math.units.inMilliseconds
+import org.team5419.fault.math.units.Ampere
+import org.team5419.fault.math.units.inAmps
+import org.team5419.fault.math.units.derived.Radian
+import org.team5419.fault.math.units.native.NativeUnitModel
+
+typealias LinearBerkeleiumSRX = BerkeliumSRX<Meter>
+typealias AngularBerkeleiumSRX = BerkeliumSRX<Radian>
+
+class BerkeliumFX<T : SIKey>(
+    val talonSRX: TalonSRX,
+    model: NativeUnitModel<T>
+) : CTREBerkeliumMotor<T>(talonSRX, model) {
+
+    constructor(id: Int, model: NativeUnitModel<T>): this(TalonSRX(id), model)
+
+    init {
+        talonSRX.configFactoryDefault(0)
+    }
+
+    fun configCurrentLimit(enabled: Boolean, config: CurrentLimitConfig? = null) {
+        talonSRX.enableCurrentLimit(enabled)
+        if (enabled && config != null) {
+            talonSRX.configContinuousCurrentLimit(config.continuousCurrentLimit.inAmps().toInt())
+            talonSRX.configPeakCurrentLimit(config.peakCurrentLimit.inAmps().toInt())
+            talonSRX.configPeakCurrentDuration(config.peakCurrentLimitDuration.inMilliseconds().toInt())
+        }
+    }
+
+    data class CurrentLimitConfig(
+        val peakCurrentLimit: SIUnit<Ampere>,
+        val peakCurrentLimitDuration: SIUnit<Second>,
+        val continuousCurrentLimit: SIUnit<Ampere>
+    )
```

Comment:
Is this Config common to multiple controller types? Can we share code?

Reply:
Yes, this is common to all CTRE controllers that support current limiting. This class should probably be in the base CTRE motor controller class.